url.js

```diff
@@ -32,10 +32,20 @@
     try {
       if (origURL) {
         nativeURL = new global.URL('http://example.com');
-        if ('searchParams' in nativeURL)
-          return;
-        if (!('href' in nativeURL))
+        if ('searchParams' in nativeURL) {
+          var url = new URL('http://example.com');
+          url.search = 'a=1&b=2';
+          if (url.href === 'http://example.com/?a=1&b=2') {
+            url.search = '';
```

Comment:
There's a problem with this.
It detects https://bugs.webkit.org/show_bug.cgi?id=162345 which is pretty minor.